### DISCUSSION TITLE
wavenumber interval

### DISCUSSION THREAD

**Discussion by MSAfify** [danielver02/ALPS#162] [Q&A] [answered]
Hi all,
Please let me know how I can control the interval of the parallel and perpendicular wavenumbers. Also, how can I obtain different roots and determine the type of instability? I need also to know how can I switch from scan kpar to kperp
Is it possible to check the polarization of the wave?
Thanks!
Mahmoud

**Comment by danielver02** [comment, marked as answer]
Hi Mahmoud,

The wave-vector interval is determined in three steps:
1) Set the initial wave vector through the variables kperp and kpar in the input file.
2) Choose a scan_type in the &scan_input_ namelist. Check the ALPS parameter reference under alps.space to see what the different numbers mean.
3) Depending on your scan_type, you can then use the parameters swi, swf, and ns to define the scan through wave-vector space.

Choosing the root to follow is done by choosing the correct "initial guess". This is done through the parameters g_om and g_gam. You may want to get an initial guess that is close to what you expect the solution to be (e.g., based on known dispersion relations or on your understanding of the instability at hand). The code will then look for the exact solution around your initial guess. Once it converges on such a solution, it will start to scan through wave-vector space.

If you don't have an idea about a good initial guess, you can also work with the map_scan method to find all solutions in a certain complex frequency range. For that, I would also refer you to the reference on alps.space that explains the steps to run a map scan. You can also look at the map test case in the test folder.

Best wishes,
-Daniel

**Reply by MSAfify** [reply]
Hi Prof. Daniel,
Thanks so much!
Mahmoud
